documentation/blog/2023-06-22-svelte-dev-overhaul.md

```diff
@@ -0,0 +1,96 @@
+
+```svelte
+<script lang="ts">
+	import type { Action } from 'svelte/action';
+
+	const foo: Action = (node) => {
+		// the node has been mounted in the DOM
+
+		return {
+			destroy() {
+				// the node has been removed from the DOM
+			}
+		};
+	};
+</script>
+
+<div use:foo />
+```
+
+## Dark mode
+
+After many years of users asking for dark mode on the website so they can read the docs for their night-time coding sessions, we finally added it! The website now has a dark mode toggle, which is also synced with your OS's dark mode settings. It can be toggled from the top navbar(bottom navbar on mobile).
+
+## Updated REPL
+
+The REPL has been rewritten from scratch to be fully typesafe, and comes with features like Dark mode, and Codemirror 6 (Which comes with many accesssibility improvements, multi-select mode, performance improvments, tree-shakability and many more features).
+
+## Redesigned homepage
+
+Is it a website redesign if the homepage doesn't get the same amount of love? 🙃
+
+Homepage has also been redesigned to resemble [kit.svelte.dev](https://kit.svelte.dev) a bit more, and features the beautiful Svelte Machine by Vedam.
+
+## Bottom navigation!
+
+We sent out a [tweet](https://twitter.com/Rich_Harris/status/1664712880791404546) about experimenting with bottom navigation bar on mobile rather than the conventional top navbar, and the response was overwhelmingly positive. So we went ahead and added it to the website! This makes it easier to navigate the website on mobile with just one hand, and also makes it easier to reach the dark mode toggle. It also shows the contents of whatever section of the site you are. For example, if you are on one of the docs page, the navbar will open by default to the docs contents. This allows for much quicker navigation between pages.
+
+If you're on mobile, you can already see it at the bottom. If you're on desktop, you can see it by resizing your browser window to a smaller size.
+
+## Unification of Svelte websites
+
+svelte.dev, kit.svelte.dev and learn.svelte.dev all use the same design system, and are now more consistent with each other. This makes it easier to navigate between the websites, and also makes it easier to maintain them. We have an internal package called `@sveltejs/site-kit`. It went through rigorous changes over last 4 months, and we have been moving all the common stuff between the sites to this package.
+
+This is the reason why learn.svelte.dev and kit.svelte.dev got the dark mode toggle before this site. We added it to site-kit for this site, and simply updated the package on those websites, getting the dark mode toggle automatically.
+
+## What's next
+
+We have many more things planned to do post-launch. Some of them are:
+
+- Redesign blog page
+- Webcontainer based REPL
```

Comment:
I don't know about this one. Web containers take too long to load. I thought we were going the other way and putting using the REPL code for the Svelte sections of learn.svelte.dev
```suggestion
```

Reply:
Ah, I see. And then `learn.svelte.dev` would use that new REPL component? The current wording made me think it would affect only https://svelte.dev/repl. Maybe it could be worded as something like:
```
- combine and standardize infrastructure for learn.svelte.dev and the REPL
```

Reply:
yes!!